src/Node.js

```diff
@@ -0,0 +1,133 @@
+  // ### DOM Node Shim
+  //
+  // This represents a superficial form of a DOM node which contains most of
+  // the properties that a regular DOM node would contain, but only the ones
+  // needed by slowparse.
+module.exports = (function(){
+  "use strict";
+
+  function Node(nodeValue, namespaceURI) {
+    var self = this;
+    self.nodeValue = nodeValue || "";
+    self.namespaceURI = namespaceURI || "";
+    self.parentNode = false;
+    self.nextSibling = false;
+    self.isVoid = false;
+    self.childNodes = [];
+    self.attributes = [];
+    Object.defineProperties(self, {
+      "innerHTML": {
+        get: function() {
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return null;
+          }
+
+          var innerHTML = "";
+
+          self.childNodes.forEach(function(childNode) {
+            innerHTML += childNode.outerHTML || "";
+          });
+
+          return innerHTML;
+        }
+      },
+      "outerHTML": {
+        get: function() {
+          if(self.nodeType === self.ATTRIBUTE_NODE) {
+            return null;
+          }
+          if(self.nodeType === self.COMMENT_NODE) {
+            return "<!--" + self.nodeValue + "-->";
+          }
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return self.nodeValue;
+          }
+
+          var attributeStr = "";
+          self.attributes.forEach(function(attr) {
+            attributeStr += attr.nodeName + "=\"" + (attr.nodeValue || "") + "\" ";
+          });
+          attributeStr = attributeStr.trim();
+          if(attributeStr.length > 0) {
+            attributeStr = " " + attributeStr;
+          }
+
+          var openingTag = "<" + self.nodeName.toLowerCase() + attributeStr + ">";
+          var closingTag = self.isVoid ? "" : "</" + self.nodeName.toLowerCase() + ">";
+
+          return openingTag + (self.innerHTML || "") + closingTag;
+        }
+      }
+    });
+  }
+
+  // Add a node to the current node as a child
+  Node.prototype.appendChild = function(node) {
+    var lastChild = this.childNodes.slice(-1)[0];
+    if(lastChild) {
+      lastChild.nextSibling = node;
+    }
+
+    node.parentNode = this;
+    this.childNodes.push(node);
+  };
+
+  Node.prototype.cloneNode = function() {
+    var node = new Node();
+    node.nodeName = this.nodeName;
+    node.nodeValue = this.nodeValue;
+    node.namespaceURI = this.namespaceURI;
+    node.nextSibling = this.nextSibling ? this.nextSibling.cloneNode() : false;
+    node.isVoid = this.isVoid;
+    this.childNodes.forEach(function(childNode) {
+      var clonedChildNode = childNode.cloneNode();
+      node.appendChild(clonedChildNode);
+    });
+    node.attributes = this.attributes.map(function(attribute) {
+      return attribute.cloneNode();
+    });
+    switch(this.nodeType) {
```

Comment:
this is just `node.nodeType = this.nodeType`, for two reasons.
1. These are constants that you're pulling from the Node.prototype, so they're always the same values. This code basically does `switch(x) { case A: node.x = A; case B: node.x = B; case C: node.x = C; etc. }`.
2. a copy constructor cannot behave differently from the original, so *all* instance values should be a straight copy.